napari/_vispy/layers/image.py

```diff
@@ -123,7 +123,10 @@ def _on_rendering_change(self) -> None:
         self._on_iso_threshold_change()
 
     def _on_colormap_change(self, event=None) -> None:
-        self.node.cmap = VispyColormap(*self.layer.colormap)
+        cmap_args = self.layer.colormap.dict()
+        cmap_args.pop('name')
+        cmap_args['bad_color'] = cmap_args.pop('nan_color')
+        self.node.cmap = VispyColormap(**cmap_args)
```

Comment:
I prefer to add proper method to our Colormap for dict construction rather than doing this in vispy.

Reply:
This method already exists (dict()). The problem here is that our `Colormap` and vispy's are diverging, and I think this is the right place to take care of this difference. Eemember we plan to have different backends, and the `_vispy` module is exactly where vispy-specific stuff should be.

Reply:
Current code makes it harder to test it. We should be able to easy iterate over all napari color maps and check if it properly converts into Vispy one. 
Without spawning viewer etc.